### PR TITLE
fix: diffs for done stores

### PR DESCRIPTION
**Pull request by alanshaw**
If an upload is "done", we should only count it as a diff to the space when it's NOT already stored in the space.

### CODE REVIEW

File: billing/test/lib/ucan-stream.js [hunk 1]
```diff
@@ -60,10 +85,40 @@ export const test = {
...
+        exists: async (s, l) => filteredReceipts.some(r => {
+          const out = /** @type {any} */ (r.out.ok)
+          return s.toString() === out.with?.toString() && out.link?.toString() === l.toString()
+        })
+      }
+    })
     assert.equal(deltas.length, receipts.length)
 
-    // ensure we have a delta for every receipt
+    // ensure we have a delta for every receipt, except
```
Comment (olizilla) [inline billing/test/lib/ucan-stream.js:121]: except...?

File: billing/lib/ucan-stream.js [hunk 2]
```diff
@@ -6,16 +6,23 @@ import * as StoreCaps from '@web3-storage/capabilities/store'
...
   const deltas = []
   for (const message of messages) {
     if (!isReceipt(message)) continue
 
     /** @type {number|undefined} */
     let size
     if (isReceiptForCapability(message, StoreCaps.add) && isStoreAddSuccess(message.out)) {
+      // If status is done, we need to check if the item is stored in _this_
+      // space or not. If it is, then there is no delta.
+      if (message.out.ok.status === 'done') {
```
Comment (olizilla) [inline billing/lib/ucan-stream.js:22]: do we increment when the space is allocated by calling `store/add`? If I keep calling `store/add` but not uploading the thing do i keep getting charged additional increments each time?

**Review by olizilla — APPROVED** [review]
Change looks good, just 1 question about how this works if i call store/add succesfully, repeatedly for the same cid, but never manage to upload that cid.

**Comment by alanshaw** [comment]
Hmm, thinking about it I think this doesn't work as I thought....

**Comment by alanshaw** [comment]
Plan is to return something `exists: boolean` in the receipt that would signal to us whether this item was _already_ stored in the space or not.

**Pull request closed without merging**